app/controllers/posts_controller.rb

```diff
@@ -9,25 +9,34 @@ def index
   end
 
   def show
+    @current_user = current_user
     @author = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
 
   def create
-    @author = User.find(params[:author_id])
+    @author = current_user
     @post = Post.new(post_params)
     @post.author = @author
```

Comment:
- [ ] <b>OPTIONAL:</b>    This would have also worked better. `@post = current_user.posts.new(post_params)`